File: frontend/src/services/quizResultService.js
import API from "./api";

export const saveQuizResult =
    async (
        fileName,
        score,
        total
    ) => {

        const response =
            await API.post(
                "/quiz-result/",
                {
                    file_name:
                        fileName,

                    score:
                        score,

                    total:
                        total
                }
            );

        return response.data;
    };

File: backend/app/services/quiz_result_service.py
from datetime import datetime

from app.database.mongodb import db


def save_quiz_result(

    current_user,

    file_name,

    score,

    total

):

    quiz_collection = (
        db["quiz_attempts"]
    )

    quiz_collection.insert_one(

        {

            "user_email":
            current_user["email"],

            "file_name":
            file_name,

            "score":
            score,

            "total":
            total,

            "date":
            datetime.now()

        }

    )

    return {

        "success": True,

        "message":
        "Quiz Result Saved"

    }